### ISSUE TITLE
The instance setup and warmup should have a timeout

### ISSUE THREAD

**Issue by facundobatista** [canonical/craft-providers#146]
IOW, we should call `buildd.BuilddBase`'s `setup` and `warmup` with a sensible timeout. E.g. 10 minutes.

Currently it waits forever, making the different tools to "stuck" or "freeze", as this [Charmcraft issue report](https://github.com/canonical/charmcraft/issues/823) shows.

**Comment by mr-cal** [comment]
Tracking with CRAFT-1475

**Comment by mr-cal** [comment]
Per a conversation with @facundobatista and @tigarmo, we discussed:

1. The current [_check_deadline()](https://github.com/canonical/craft-providers/blob/c6e624516dbb76ff60abfe9a2459bffe288d5e0e/craft_providers/bases/buildd.py#L70) function isn't used
    - Even if it was used, it doesn't add much value.
    - There's no good deadline to choose, since setup is highly dependent on network connectivity and computer resources.  (10 minutes would be a good deadline for my computer, but 1 hour may be appropriate for someone else)
2. _If_ craft-providers were to timebox the setup, it should be fine-grained way.  
    - We could set short deadlines for simple operations (e.g. updating the contents of a file) and very long deadlines for uncertain operations (e.g. `apt install`)
    - This could be done by passing a new `timeout` parameter to [`_execute_run()`](https://github.com/canonical/craft-providers/blob/c6e624516dbb76ff60abfe9a2459bffe288d5e0e/craft_providers/bases/buildd.py#L108) with a timeout value.
3. Every issue I've seen regarding the setup getting stuck involve the [`_setup_wait_for_network()`](https://github.com/canonical/craft-providers/blob/c6e624516dbb76ff60abfe9a2459bffe288d5e0e/craft_providers/bases/buildd.py#L896):
```python
command = ["getent", "hosts", "snapcraft.io"]
while True:
    proc = _execute_run(executor, command, check=False)
```


#### Resolution
1. Remove the unused `_check_deadline()`
2. Replace the `while True:` block above with a 2 minute timeout
3. If we start receiving issues about the setup getting stuck and not timing out, then we can consider adding the `_execute_run(timeout)` feature described above

**Comment by mr-cal** [comment]
Resolved with #274 and #305
